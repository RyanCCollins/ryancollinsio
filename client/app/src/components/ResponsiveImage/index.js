import React, { PropTypes, Component } from 'react';
import cssModules from 'react-css-modules';
import Animate from 'grommet-udacity/components/Animate';
import MissingVisual from './missing-visual.png';
import styles from './index.module.scss';

class ResponsiveImage extends Component {
  constructor() {
    super();
    this.setImageLoadedState = this.setImageLoadedState.bind(this);
    this.state = {
      isLoaded: false,
    };
  }
  setImageLoadedState() {
    this.setState({
      isLoaded: true,
    });
  }
  render() {
    const {
      src,
      matchHeight,
      className,
    } = this.props;
    return (
      <Animate
        visible={this.state.isLoaded}
        enter={{ animation: 'fade', duration: 1000 }}
        keep
      >
        <div
          className={matchHeight ? styles.responsiveImageMatchHeight : styles.responsiveImage}
        >
          <img
            alt="responsive"
            className={className}
            onLoad={this.setImageLoadedState}
            src={src || MissingVisual}
          />
        </div>
      </Animate>
    );
  }
}

ResponsiveImage.propTypes = {
  src: PropTypes.string.isRequired,
  matchHeight: PropTypes.bool.isRequired,
  className: PropTypes.string.isRequired,
};

ResponsiveImage.defaultProps = {
  matchHeight: true,
  className: 'img',
};

export default cssModules(ResponsiveImage, styles);
