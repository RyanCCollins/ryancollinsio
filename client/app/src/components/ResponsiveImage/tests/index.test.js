import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import ResponsiveImage from '../index';

describe('<ResponsiveImage />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <ResponsiveImage />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
