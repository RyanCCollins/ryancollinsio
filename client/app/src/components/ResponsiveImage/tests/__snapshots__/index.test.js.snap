exports[`<ResponsiveImage /> should render with default props 1`] = `
<Animate
  enter={
    Object {
      "animation": "fade",
      "duration": 1000
    }
  }
  keep={true}
  visible={false}>
  <div>
    <img
      alt="responsive"
      className="img"
      onLoad={[Function bound setImageLoadedState]}
      src={Object {}} />
  </div>
</Animate>
`;
